presto-spi/src/test/java/io/prestosql/spi/block/BaseBlockEncodingTest.java

```diff
@@ -15,17 +15,80 @@
 
 import io.airlift.slice.DynamicSliceOutput;
 import io.prestosql.spi.type.Type;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Random;
+import java.util.stream.IntStream;
 
 import static io.prestosql.spi.block.BlockTestUtils.assertBlockEquals;
 
 public abstract class BaseBlockEncodingTest<T>
 {
+    private static final int[] RANDOM_BLOCK_SIZES = {2, 4, 8, 9, 16, 17, 32, 33, 64, 65, 1000, 1000000};
+
+    private final Random random = new Random(32167);
     private final BlockEncodingSerde blockEncodingSerde = new TestingBlockEncodingSerde();
 
     protected abstract Type getType();
 
     protected abstract void write(BlockBuilder blockBuilder, T value);
 
+    protected abstract T randomValue(Random random);
+
+    @Test
+    public void testEmpty()
+    {
+        roundTrip();
+    }
+
+    @Test
+    public void testSingleNull()
+    {
+        roundTrip((T) null);
+    }
+
+    @Test
+    public void testSingleValue()
+    {
+        roundTrip(randomValue(random));
+    }
+
+    @Test(dataProvider = "testRandomDataDataProvider")
+    public void testRandomData(int size, BlockFill fill)
```

Comment:
Do we expect this randomized test to find any issues? Do we need all these size combinations? It seems that few purposefully designed cases are enough.

Reply:
One of the advantages of those tests is that they are not expensive at all. Running hundreds of them will take less than a second.
I expect any actual case to be treated as a separate test

Reply:
We need to make tests to be single threaded, otherwise we might hit some issues with reproducing GHA tests execution flakiness.

Maybe use rangom that is bound to method execution life cycle, not a field of the class

Reply:
These tests are super fast. There is no harm in running them single-threaded. I will add the annotation
Making random method-bound is not trivial here.